tests/js/set-tests.js

```diff
@@ -154,37 +188,36 @@ module.exports = {
 
     //
     // test filtering on object properties
-    async testSetObjectFilter() {
+    testSetObjectFilter() {
         const personSchema = {
             name: "Person",
             properties: {
-                firstName: 'string',
-                age: 'int'
+                firstName: "string",
+                age: "int"
             }
         };
 
         const teamSchema = {
-            name: 'Team',
+            name: "Team",
             properties: {
-                persons: 'Person<>'
+                persons: "Person<>"
             }
         };
 
         const realm = new Realm({ schema: [personSchema, teamSchema] });
-        const schema = realm.schema;
 
         realm.write(() => {
-            // insert two people
+            // insert three people
             realm.create(teamSchema.name, {persons: [
                 {firstName: "Joe",  age: 4},
                 {firstName: "Sue",  age: 53},
-                {firstName: 'Bob', age: 99},
+                {firstName: "Bob", age: 99},
             ]});
         });
 
 
         let teams = realm.objects(teamSchema.name);
-        let filteredSues = teams[0].persons.filtered('firstName = "Sue"');
+        let filteredSues = teams[0].persons.filtered("firstName = \"Sue\"");
```

Comment:
Find it slightly annoying that these needs to be escaped, can we just use single-quotes in the query?
```suggestion
        let filteredSues = teams[0].persons.filtered("firstName = 'Sue'");
```

or alternatively a placeholder

```suggestion
        let filteredSues = teams[0].persons.filtered("firstName = $0", "Sue");
```